x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -203,12 +203,15 @@ ResolvedIndices resolveIndicesAndAliases(IndicesRequest indicesRequest, MetaData
             } else {
                 resolvedIndicesBuilder.addLocal(aliasesRequest.aliases());
             }
-            // if no aliases are authorized, then fill in an expression that
-            // MetaData#findAliases evaluates to the empty alias list. You cannot put
-            // "nothing" (the empty list) explicitly because this is resolved by es core to
-            // _all
-            if (aliasesRequest.aliases().length == 0) {
-                aliasesRequest.replaceAliases(NO_INDICES_OR_ALIASES_ARRAY);
+            /*
+             * If no aliases are authorized for a get aliases request, then fill in an expression that MetaData#findAliases evaluates to an
+             * empty alias list. We can not put an empty list here because core resolves this as _all. For other request types, this
+             * replacement is not needed and can trigger issues when we rewrite the request on the coordinating node. For example, for a
+             * remove index request, if we did this replacement, the request would be rewritten to include "*", "-*" and for a user that
+             * does not have permissions on "*", the master node would not authorize the request.
+             */
+            if (indicesRequest instanceof GetAliasesRequest && aliasesRequest.aliases().length == 0) {
+                 aliasesRequest.replaceAliases(NO_INDICES_OR_ALIASES_ARRAY);
```

Comment:
This seems to change the behaviour for `remove` alias, because it [does support wildcards](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/admin/indices/alias/IndicesAliasesRequest.java#L455), and when it [expands aliases](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java#L170) it interprets [empty as all](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java#L337).

Reply:
Nice catch Tim! I have pushed the changes such that all aliases request types that expand alias parameters (i.e. `remove alias` and `get alias`) will replace any empty lists (because no aliases are authorized for the user) with the `"*","-*"` pattern.

Reply:
Nice catch @tvernum.